src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -385,48 +405,24 @@ internal static class ConstraintsHelper
 
             // Nullability checks can only add warnings here so skip them for this check as we are only
             // concerned with errors.
-            CheckAllConstraints(type, compilation, conversions, includeNullability: false, NoLocation.Singleton, diagnostics);
+            type.CheckAllConstraints(new CheckConstraintsArgs(compilation, conversions, includeNullability: false, NoLocation.Singleton, diagnostics));
             bool ok = !diagnostics.HasAnyErrors();
             diagnostics.Free();
             return ok;
         }
 
-        public static void CheckAllConstraints(
-            this TypeSymbol type,
-            CSharpCompilation compilation,
-            ConversionsBase conversions,
-            bool includeNullability,
-            Location location,
-            DiagnosticBag diagnostics)
+        public static void CheckAllConstraints(this TypeSymbol type, CheckConstraintsArgs args)
         {
-            type.VisitType(s_checkConstraintsSingleTypeFunc, new CheckConstraintsArgs(compilation, conversions, includeNullability, location, diagnostics));
-        }
-
-        private readonly struct CheckConstraintsArgs
-        {
-            public readonly CSharpCompilation CurrentCompilation;
-            public readonly ConversionsBase Conversions;
-            public readonly bool IncludeNullability;
-            public readonly Location Location;
-            public readonly DiagnosticBag Diagnostics;
-
-            public CheckConstraintsArgs(CSharpCompilation currentCompilation, ConversionsBase conversions, bool includeNullability, Location location, DiagnosticBag diagnostics)
-            {
-                this.CurrentCompilation = currentCompilation;
-                this.Conversions = conversions;
-                this.IncludeNullability = includeNullability;
-                this.Location = location;
-                this.Diagnostics = diagnostics;
-            }
+            type.VisitType(s_checkConstraintsSingleTypeFunc, args);
         }
 
         private static readonly Func<TypeSymbol, CheckConstraintsArgs, bool, bool> s_checkConstraintsSingleTypeFunc = (type, arg, unused) => CheckConstraintsSingleType(type, arg);
 
-        private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraintsArgs args)
+        private static bool CheckConstraintsSingleType(TypeSymbol type, in CheckConstraintsArgs args)
         {
             if (type.Kind == SymbolKind.NamedType)
             {
-                ((NamedTypeSymbol)type).CheckConstraints(args.CurrentCompilation, args.Conversions, args.IncludeNullability, args.Location, args.Diagnostics);
+                ((NamedTypeSymbol)type).CheckConstraints(args);
```

Comment:
>args [](start = 57, length = 4)

Would it make sense to pass it by ref here as well?

Reply:
The argument is passed by reference - the `CheckConstraints()` parameter is `in`.

---
In reply to: [405099677](https://github.com/dotnet/roslyn/pull/43094#discussion_r405099677) [](ancestors = 405099677)

Reply:
nit: consider using an explicit 'in' where possible for clarity.